Migrate remainder of THCDeviceUtils.cuh to ATen

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65492
* #65491
* __->__ #65472
* #65471
* #65470

Differential Revision: [D31148181](https://our.internmc.facebook.com/intern/diff/D31148181)
BC-breaking note:
This PR remove THCeilDiv function and THC/THCDeviceUtils.cuh header, please use at::ceil_div and ATen/ceil_div.h header as appropriate

peterbell10 committed:
Migrate remainder of THCDeviceUtils.cuh to ATen

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate remainder of THCDeviceUtils.cuh to ATen"




[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate remainder of THCDeviceUtils.cuh to ATen"




[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate remainder of THCDeviceUtils.cuh to ATen"




[ghstack-poisoned]

ngimel commented on aten/src/ATen/ceil_div.h:
```diff
@@ -0,0 +1,23 @@
+   Computes ceil(a / b)
+*/
+template <typename T, typename=std::enable_if_t<std::is_integral<T>::value>>
+C10_ALWAYS_INLINE C10_HOST_DEVICE T ceil_div(T a, T b) {
```
There's already ATenCeilDiv in CUDAApplyUtils.cuh already, can you use it? And add `round_up` to that header.

peterbell10 replied:
I've done it the other way around, replacing ATenCeilDiv with `ceil_div`. My reasoning is that this isn't cuda specific, and would be useful to have available for future CPU code. Also, that including all of `CUDAApplyUtils.cuh` just for a small helper function is unnecessary.

peterbell10 committed:
Update on "Migrate remainder of THCDeviceUtils.cuh to ATen"




[ghstack-poisoned]

peterbell10 committed:
Remove ATenCeilDiv in favour of ceil_div on "Migrate remainder of THCDeviceUtils.cuh to ATen"




[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate remainder of THCDeviceUtils.cuh to ATen"




[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148181).

peterbell10 committed:
Remove manual hipifying include path on "Migrate remainder of THCDeviceUtils.cuh to ATen"


Differential Revision: [D31148181](https://our.internmc.facebook.com/intern/diff/D31148181)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148181).

ngimel commented:
Torchvision unfortunately is using ATenCeilDiv https://github.com/pytorch/vision/blob/7947fc8fb38b1d3a2aca03f22a2e6a3caa63f2a0/torchvision/csrc/ops/cuda/interpolate_aa_kernels.cu#L266, for some reason this wasn't caught by the tests

peterbell10 committed:
Update on "Migrate remainder of THCDeviceUtils.cuh to ATen"


Differential Revision: [D31148181](https://our.internmc.facebook.com/intern/diff/D31148181)

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate remainder of THCDeviceUtils.cuh to ATen"


Differential Revision: [D31148181](https://our.internmc.facebook.com/intern/diff/D31148181)

[ghstack-poisoned]

peterbell10 committed:
Update on "Migrate remainder of THCDeviceUtils.cuh to ATen"


Differential Revision: [D31148181](https://our.internmc.facebook.com/intern/diff/D31148181)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148181).

ngimel commented:
A lot of THCeilDivs internally, I'm fixing them, will land on Monday.
